src/internal/operators/share.ts

```diff
@@ -132,7 +132,8 @@ export function share<T>(options?: ShareConfig<T>): OperatorFunction<T, T> {
 
     return () => {
       refCount--;
-      castSubscription.unsubscribe();
+      castSubscription!.unsubscribe();
+      castSubscription = null;
```

Comment:
If I uncomment this line, the added test still passes. Not sure what I'm missing.